packages/docs/src/content/docs/reference/faq.md

```diff
@@ -0,0 +1,326 @@
+entry files recursively until no more entry files are found.
+
+## Why does Knip have plugins?
+
+Plugins are an essential part of Knip. They save a lot of configuration out of
+the box by adding entry files as accurately as possible and only for the tools
+actually installed. Yet the real magic is in their custom parsers for
+configuration files.
+
+For instance, Vitest has the `environment` configuration option. The Vitest
+plugin knows `"node"` is the default value for `environment` which does not
+require an extra package, but will translate `"edge-runtime"` to the
+`@edge-runtime/vm` package. This allows Knip to report it if this package is not
+listed in `package.json`, or when it is no longer used after changes in the
+Vitest configuration.
+
+Configuration files may also contain references to entry files. For instance,
+Jest has `setupFilesAfterEnv: "<rootDir>/jest.setup.js"` or a reference may
+point to a file in another workspace in the same monorepo, e.g.
+`setupFiles: ['@org/shared/jest-setup.ts']`. Those entry files may also contain
+imports of internal modules or external dependencies, and so on.
+
+With an easy-to-use plugin API, many plugins have been created by contributors.
+
+## Why is Knip so heavily engineered?
+
+Even though I love the Unix philosophy, at this point I believe for Knip it
+makes sense to have the pieces in a single tool.
+
+Building up the module and dependencies graph requires non-standard module
+resolution and not only static but also dynamic analysis (i.e. actually load and
+execute modules), such as for parsers of plugins to receive the exported value
+of dynamic tooling configuration files. Additionally, [exports consumed by
+external libraries][6] require type information, as supported by the TypeScript
+backend.
+
+The rippling effect of plugins and recursively adding entry files and
+dependencies to build up this graph - as answered with the previous question -
+is also exactly what's meant by ["comprehensive" here][7].
+
+## Does Knip handle Vue or Astro files?
+
+To further increase the coverage of the module graph, non-standard files other
+than JavaScript and TypeScript modules should be included as well. For instance,
+`.mdx` and `.astro` files can import each other, internal modules and external
+dependencies.
+
+Knip includes basic "compilers" for a few common file types (Astro, MDX, Svelte,
+Vue). Knip does not include actual compilers for reasons of potential
+incompatibility with the existing compiler, and dependency size. Knip allows to
+override them with the compiler(s) in your project, and add additional ones for
+other file types.
+
+## Why are the exports of my `.svelte` files not used?
+
+Knip comes with basic "compilers" for a few common non-standard file types.
+They're not actual compilers, they're regular expressions only to extract import
+statements. Override the built-in Svelte "compilers" with the real one in your
```

Comment:
I think that the suggested solution probably deserves a bit more hold-handing/explanations (unless you believe it's clear enough). A link to a more comprehensive solution is just as fine as well.

Reply:
Yeah, there's a link to https://knip.dev/features/compilers - this FAQ item is merely to illustrate the main idea.